pkg/cli/repositories.go

```diff
@@ -0,0 +1,447 @@
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		name := args[0]
+		key := args[1]
+
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+
+		// Check if file exists
+		if !flagForce {
+			// if stat returns no error return error saying you need to provide the force flag
+			_, err := os.Stat(filepath.Join(pathCheck, name))
+			if err == nil {
+				SetError(errors.New("key file already exists provide --force to overwrite"), 2)
+				return
+			}
+		}
+
+		// Write key to a file under that keys directory
+		err = ioutil.WriteFile(filepath.Join(pathCheck, name), []byte(key), os.ModePerm)
+		if err != nil {
+			SetError(err, 3)
+			return
+		}
+
+		// Check if default and write another file called default under repository-keys
+		if flagDefault {
+			err = ioutil.WriteFile(filepath.Join(pathCheck, "default"), []byte(key), os.ModePerm)
+			if err != nil {
+				SetError(err, 4)
+				return
+			}
+		}
+	},
+}
+
+func init() {
+	f := createKeyCmd.Flags()
+	f.BoolVar(&flagDefault, "default", false, "save this key to use as default")
+	f.BoolVar(&flagForce, "force", false, "force overwrite of key file")
+}
+
+var listKeysCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List all keys currently stored",
+	Args:  cobra.MaximumNArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+
+		if _, err = os.Stat(pathCheck); os.IsNotExist(err) {
+			fmt.Printf("no keys found\n")
+			return
+		}
+
+		fis, err := ioutil.ReadDir(pathCheck)
+		if err != nil {
+			SetError(err, 2)
+			return
+		}
+
+		for _, fi := range fis {
+			if fi.Name() != "default" {
+				fmt.Println(fi.Name())
+			}
+		}
+	},
+}
+
+var deleteKeyCmd = &cobra.Command{
+	Use:   "delete NAME",
+	Short: "Delete a key currently stored",
+	Args:  cobra.MaximumNArgs(1),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return errors.New("Must provide the name of the key you want to delete")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		name := args[0]
+
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+		path := filepath.Join(pathCheck, name)
+		dpath := filepath.Join(pathCheck, "default")
+
+		// before removing we should check if default is the same and delete that
+		f1, err := ioutil.ReadFile(path)
+		if err != nil {
+			SetError(fmt.Errorf("%s keyfile does not exist", name), 2)
+			return
+		}
+
+		f2, err := ioutil.ReadFile(dpath)
+		if err != nil {
+			if !errors.Is(err, os.ErrNotExist) {
+				SetError(err, 3)
+				return
+			}
+		}
+
+		// If same bytes remove default aswell
+		if bytes.Equal(f1, f2) {
```

Comment:
If you run `$ borteil repositories keys delete default` the `default` key will be deleted here and then it will try to delete itself again in line 240 which will lead to an error